packages/cheqd/src/anoncreds/services/CheqdAnonCredsRegistry.ts

```diff
@@ -349,7 +421,79 @@ export class CheqdAnonCredsRegistry implements AnonCredsRegistry {
     }
   }
 
-  public async registerRevocationStatusList(): Promise<RegisterRevocationStatusListReturn> {
-    throw new Error('Not implemented!')
+  public async registerRevocationStatusList(
+    agentContext: AgentContext,
+    { revocationStatusList, options }: RegisterRevocationStatusListOptions
+  ): Promise<RegisterRevocationStatusListReturn> {
+    try {
+      const revocationRegistryDefinition = await this.getRevocationRegistryDefinition(
+        agentContext,
+        revocationStatusList.revRegDefId
+      )
+      if (!revocationRegistryDefinition.revocationRegistryDefinition) {
+        throw new CredoError(`Revocation registry definition not found for id: ${revocationStatusList.revRegDefId}`)
+      }
+
+      const revocationRegistryDefinitionName = revocationRegistryDefinition.revocationRegistryDefinitionMetadata.name
+      if (!revocationRegistryDefinitionName)
+        throw new CredoError(
+          `Revocation registry definition name not found for id: ${revocationStatusList.revRegDefId}`
+        )
+
+      const cheqdDidRegistrar = agentContext.dependencyManager.resolve(CheqdDidRegistrar)
+      const revocationStatusListResource = {
+        id: utils.uuid(),
+        name: revocationRegistryDefinitionName as string,
+        resourceType: 'anonCredsStatusList',
+        data: {
+          currentAccumulator: revocationStatusList.currentAccumulator,
+          revRegDefId: revocationStatusList.revRegDefId,
+          revocationList: revocationStatusList.revocationList,
+        },
+        version: utils.uuid(),
+      } satisfies CheqdCreateResourceOptions
+
+      const response = await cheqdDidRegistrar.createResource(
+        agentContext,
+        revocationStatusList.issuerId,
+        revocationStatusListResource
+      )
+      if (response.resourceState.state !== 'finished')
+        throw new CredoError(response.resourceState.reason ?? 'Unknown error')
+
+      // It's not possible to get the timestamp from the response, so we set it to the current time
```

Comment:
Why not? The timestamp is important for some pieces, so I think we need to get the timestamp is will be used when you resolve the status list (then we can get the timestamp using response.resourceMetadata.created.getTime()?)

Reply:
The cheqd sdk currently doesn't return the creation date when you create a new resource, but it's returned when you are querying a resource. So the only way is to query it just after creation? Also when you look at the raw response of the cheqd sdk of the creation it doesn't return a creation.

But for fetching the statuslist it only needs a timestamp which is close to the resource it doesn't really have to be exact.

So what do you think we should do in this case?